### ISSUE TITLE
Settting frame.axis can flip display 180 degrees

### ISSUE THREAD

**Issue by BruceSherwood** [BruceSherwood/vpython-wx#55]
Reported by Geoff Tovey, bug in setting frame.axis (display flips in some orientations):

from visual import *

f = frame()
box() # marks frame origin (centre of rotation)

c = curve(frame=f, pos=[(0,0,0),(8,0,0),
(8,0,4),(8,2,4)], color=color.cyan)

while True:
for angle in range(0,360):
rate(100)
angleR = radians(angle)
f.axis = ( cos(angleR), sin(angleR), 0) # this fails
f.rotate(angle=radians(1), axis=(0,0,1)) # this works
if f.frame_to_world( (0,0,10)).z < 0:
c.color = color.red
else:
c.color = color.cyan

**Comment by ndjohnson** [comment]
Bruce - I managed to solve my particular issue, though I’m not entirely sure how.  I took your advice about avoiding manipulating frame.up and took more care about initialising frame.up before each rotation of frame.axis about itself, and it stopped the axis flipping I was observing.  Not sure in the end whether I was uncovering a bug, or just programming it wrongly.  Most likely the latter, I suspect.  I’ll try it again with 6.11 when you issue it, and take it from there.

Thanks again for the insight, and have a great 2015

Best

N.
